enterprise/backend/src/metabase_enterprise/sso/integrations/saml.clj

```diff
@@ -109,14 +111,24 @@
   (api/check (sso-settings/saml-enabled)
     [400 (tru "SAML has not been enabled and/or configured")]))
 
+(defn- has-host? [uri]
+  (try
+    (-> uri URI. .getHost some?)
+    (catch MalformedURLException _ false)))
+
 (defmethod sso.i/sso-get :saml
   ;; Initial call that will result in a redirect to the IDP along with information about how the IDP can authenticate
   ;; and redirect them back to us
   [req]
   (check-saml-enabled)
-  (let [redirect-url (or (get-in req [:params :redirect])
+  (let [redirect (get-in req [:params :redirect])
+        redirect-url (if (nil? redirect)
+                       (do
                          (log/warn (trs "Warning: expected `redirect` param, but none is present"))
-                         (public-settings/site-url))]
+                         (public-settings/site-url))
+                       (if (has-host? redirect)
+                         redirect
+                         (str (public-settings/site-url) redirect)))]
```

Comment:
Can it happen that this constructs a URI which results in a `URISyntaxException` in `sso-utils/check-sso-redirect`? And if it does, will that be handled correctly?